src/common/selftest/selftest_fan.h

```diff
@@ -0,0 +1,51 @@
+// selftest_fan.h
+#pragma once
+
+#include <inttypes.h>
+#include "selftest_MINI.h"
+#include "fanctl.h"
+
+typedef struct _selftest_fan_config_t {
+    const char *partname;
+    CFanCtl *pfanctl;
+    int pwm_start;
+    int pwm_step;
+    int pwm_steps;
+} selftest_fan_config_t;
+
+class CSelftestPart_Fan : public CSelftestPart {
+public:
+    enum TestState : uint8_t {
+        spsIdle,
+        spsStart,
+        spsWait_stopped,
+        spsWait_rpm,
+        spsMeasure_rpm,
+        spsFinish,
+        spsFinished,
+        spsAborted,
+    };
+
+public:
+    CSelftestPart_Fan(const selftest_fan_config_t *pconfig);
+
+public:
+    bool IsInProgress() const;
+
+public:
+    bool Start();
```

Comment:
parent has this method virtual, so this one is virtual too. Use virtual and override so it is obvious
virtual  bool Start() override;

Reply:
fixed